### PR TITLE
COORDINATIONS: Identification Orchestration - Process Identification Requests

### PR DESCRIPTION
**Pull request by MichaelOSullivanAnswer**

### CODE REVIEW

File: ISL.Reidentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.cs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+                service.ValidateAccessForIdentificationRequestsAsync(inputAccessRequest))
+                    .ReturnsAsync(outputOrchestrationAccessRequest);
+
+            this.identificationOrchestrationServiceMock.Setup(service =>
+                service.ProcessIdentificationRequestAsync(outputOrchestrationAccessRequest.IdentificationRequest))
+                    .ReturnsAsync(outputIdentificationRequest);
+
+            // when
+            var actualAccessRequest =
+                await this.identificationCoordinationService.ProcessIdentificationRequestsAsync(inputAccessRequest);
```
Comment (cjdutoit) [inline ISL.Reidentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.cs:38]:
```suggestion
            AccessRequest actualAccessRequest = await this.identificationCoordinationService
                .ProcessIdentificationRequestsAsync(inputAccessRequest);
```

Comment (MichaelOSullivanAnswer) [reply ISL.Reidentification.Core.Tests.Unit/Services/Coordinations/Identifications/IdentificationCoordinationTests.Logic.cs]: This has been updated

File: ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs [hunk 2]
```diff
@@ -29,6 +29,21 @@ public ValueTask<AccessRequest> ProcessDelegatedAccessRequestAsync(AccessRequest
             throw new System.NotImplementedException();
 
         public ValueTask<AccessRequest> ProcessIdentificationRequestsAsync(AccessRequest accessRequest) =>
-            throw new System.NotImplementedException();
+        TryCatch(async () =>
+        {
+            ValidateAccessRequestIsNotNull(accessRequest);
+            ValidateIdentificationRequestIsNotNull(accessRequest.IdentificationRequest);
```
Comment (cjdutoit) [inline ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs:35]: Please wrap both these calls in a method on the validation file.  This prevent clutter on this method and it also helps to see what validation is done where if all that is on the validation file.

ValidateOnProcessIdentificationRequests(accessRequest)
```suggestion
            ValidateOnProcessIdentificationRequests(accessRequest)
```

Comment (MichaelOSullivanAnswer) [reply ISL.ReIdentification.Core/Services/Coordinations/Identifications/IdentificationCoordinationService.cs]: This has been updated

**Review by cjdutoit — APPROVED** [review]

**Pull request merged**